src/ring_theory/subring/basic.lean

```diff
@@ -651,6 +651,40 @@ lemma center.coe_div (a b : center K) : ((a / b : center K) : K) = (a : K) / (b
 
 end division_ring
 
+section centralizer
+
+/-- The centralizer of a ring as a `subring`. -/
+def centralizer (s : set R) : subring R :=
+{ neg_mem' := λ x, set.neg_mem_centralizer,
+  ..subsemiring.centralizer s }
+
+@[simp, norm_cast]
+lemma coe_centralizer (s : set R) : (centralizer s : set R) = s.centralizer := rfl
+
+lemma centralizer_to_submonoid (s : set R) :
+  (centralizer s).to_submonoid = submonoid.centralizer s := rfl
+
+lemma centralizer_to_subsemiring (s : set R) :
+  (centralizer s).to_subsemiring = subsemiring.centralizer s := rfl
+
+lemma mem_centralizer_iff {s : set R} {z : R} :
+  z ∈ centralizer s ↔ ∀ g ∈ s, g * z = z * g :=
+iff.rfl
+
+lemma center_le_centralizer (s) : center R ≤ centralizer s := s.center_subset_centralizer
+
+lemma centralizer_le (s t : set R) (h : s ⊆ t) : centralizer t ≤ centralizer s :=
+set.centralizer_subset h
+
+@[simp] lemma centralizer_univ : centralizer set.univ = center R :=
+set_like.ext' (set.centralizer_univ R)
+
+lemma mem_center_of_centralizer_eq_top {r : R} (h : centralizer ({r} : set R) = ⊤) :
+  r ∈ center R :=
+λ x, eq.symm $ by simpa [mem_centralizer_iff] using eq_top_iff.mp h (mem_top x)
+
+end centralizer
```

Comment:
Maybe we could have a simp lemma that the centralizer of any set is the top when we have a comm_ring, assuming simp can't prove that already. Or the above `mem_center_of_centralizer_eq_top` could be simp too which would probably do it

Reply:
sadly not enough because it's only about singleton sets - I added the lemma though! I was trying to match the API for the other `centralizer`s bar the extra stuff that is in the Wedderburn API, at some point upgrading those to feature parity would be nice too

Reply:
Yeah makes sense, maybe one of erics suggstions would help then.
Maybe you could add a TODO in the code somewhere to unify and keep in sync the APIs? It might be less likely to be forgotten that way

Reply:
in the end I think I will just end up syncing them using Eric's suggestions